src/ort_genai_c.cpp

```diff
@@ -208,6 +208,15 @@ OgaResult* OGA_API_CALL OgaGenerator_GenerateNextToken(OgaGenerator* generator)
   OGA_CATCH
 }
 
+OgaResult* OGA_API_CALL OgaGenerator_GetOutput(const OgaGenerator* oga_generator, const char* name, OgaTensor** out) {
+  OGA_TRY
+  auto& generator = *reinterpret_cast<const Generators::Generator*>(oga_generator);
+  auto* ortValueOutput = generator.state_->GetOutput(name);
+  *out = reinterpret_cast<OgaTensor*>(ortValueOutput);
```

Comment:
The buffer of 'OgaTensor** out' is not handled properly here. Its buffer is shared with the buffer of output tensors of the model. The lifetime of 'OgaTensor** out' can be different with the output tensor. 

We need to change OgaTensor to make it able to own the buffer.

Reply:
Thanks, fixed.